core/src/main/java/com/amplifyframework/analytics/UserProfile.java

```diff
@@ -0,0 +1,313 @@
+        @Override
+        public int hashCode() {
+            int result = latitude != null ? latitude.hashCode() : 0;
+            result = 31 * result + (longitude != null ? longitude.hashCode() : 0);
+            result = 31 * result + (postalCode != null ? postalCode.hashCode() : 0);
+            result = 31 * result + (city != null ? city.hashCode() : 0);
+            result = 31 * result + (region != null ? region.hashCode() : 0);
+            result = 31 * result + (country != null ? country.hashCode() : 0);
+            return result;
+        }
+
+        @Override
+        public String toString() {
+            return "Location{" +
+                    "latitude=" + latitude +
+                    ", longitude=" + longitude +
+                    ", postalCode='" + postalCode + '\'' +
+                    ", city='" + city + '\'' +
+                    ", region='" + region + '\'' +
+                    ", country='" + country + '\'' +
+                    '}';
+        }
+
+        public static final class Builder {
+            private Double latitude;
+            private Double longitude;
+            private String postalCode;
+            private String city;
+            private String region;
+            private String country;
+
+            public Builder latitude(@NonNull Double latitude) {
+                Objects.requireNonNull(latitude);
+                this.latitude = latitude;
+                return this;
+            }
+
+            public Builder longitude(@NonNull Double longitude) {
+                Objects.requireNonNull(longitude);
+                this.longitude = longitude;
+                return this;
+            }
+
+            public Builder postalCode(@NonNull String postalCode) {
+                Objects.requireNonNull(postalCode);
+                this.postalCode = postalCode;
+                return this;
+            }
+
+            public Builder city(@NonNull String city) {
+                Objects.requireNonNull(city);
+                this.city = city;
+                return this;
+            }
+
+            public Builder region(@NonNull String region) {
+                Objects.requireNonNull(region);
+                this.region = region;
+                return this;
+            }
+
+            public Builder country(@NonNull String country) {
+                Objects.requireNonNull(country);
+                this.country = country;
+                return this;
+            }
+
+            public Location build() {
+                if (latitude == null &&
+                    longitude == null &&
+                    postalCode == null &&
+                    city == null &&
+                    region == null &&
+                    country == null) {
+                    throw new IllegalArgumentException("At least on parameter must be non-null to " +
+                            "construct a Location object");
+                }
```

Comment:
Since the check is so unlikely to trigger, it may not be worth it to maintain this code. Might be better off just allowing an object with all `null`s`.